build_tools/cmake/iree_copts.cmake

```diff
@@ -335,10 +335,10 @@ if(CMAKE_CXX_FLAGS AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
 endif()
 
 if(NOT ANDROID AND IREE_ENABLE_THREADING)
-  iree_select_compiler_opts(_IREE_PTHREADS_LINKOPTS
-    CLANG_OR_GCC
-      "-lpthread"
-  )
+  set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+  set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+  find_package(Threads REQUIRED)
+  link_libraries(Threads::Threads)
```

Comment:
I'm worried this could adversely affect other projects that include IREE as a dependency.

We could instead add `Threads::Threads` in `iree_cc_library` to keep the same behavior.

https://github.com/openxla/iree/blob/63fee4852bfa89e9b3bd84514092950ec9951a37/build_tools/cmake/iree_cc_library.cmake#L188-L196

(untested ~pseudocode)
In this file:
```cmake
# if threading enabled and found, otherwise set to empty
set(IREE_THREADS_DEPS Threads::Threads)
```
In `iree_cc_library.cmake` where I linked:
```cmake
    # Also depend in Threads (-pthread) as needed
    target_link_libraries(${_NAME}
      PUBLIC
        ${_RULE_DEPS}
        IREE_THREADS_DEPS 
    )
```

Reply:
Sure, I'll move it to `iree_cc_library.cmake`. It sounds like getting pthread library is not compulsory, so I'll remove the `REQUIRED` keyword in `find_package`.

I read this line of code:
```
  # Implicit deps. line 113
  if(IREE_IMPLICIT_DEFS_CC_DEPS)
    list(APPEND _RULE_DEPS ${IREE_IMPLICIT_DEFS_CC_DEPS})
  endif()
```
So I guess it is alright for me to add `Threads::Threads` into `RULES_DEPS`, right?

Like this:
```
# Right after appending implicit deps

  # Find and add threads as dependency.
  if(NOT ANDROID AND IREE_ENABLE_THREADING)
    set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
    set(THREADS_PREFER_PTHREAD_FLAG TRUE)
    find_package(Threads)
    list(APPEND _RULE_DEPS Threads::Threads)
  else()
    # Android provides its own pthreads support with no linking required.
  endif()
```